### PR TITLE
add robust handling of non-UTF8 when detecting inventory/playbooks

### PR DESCRIPTION
**Pull request by ryanpetrello**
We shouldn't assume that these files are UTF-8 encoded.  If we encounter files in SCM that we can't decode, then just skip them because they're probably not playbooks or inventories.

### CODE REVIEW

File: awx/main/utils/ansible.py [hunk 1]
```diff
@@ -44,7 +45,12 @@ def could_be_playbook(project_path, dir_path, filename):
     # show up.
     matched = False
     try:
-        for n, line in enumerate(open(playbook_path)):
+        for n, line in enumerate(codecs.open(
+            playbook_path,
+            'r',
+            encoding='utf-8',
+            errors='ignore'
```
Comment (ryanpetrello) [inline awx/main/utils/ansible.py:52]: Just ignore anything we can't decode; the purpose of these functions is just to look for certain regular expression matches in the files.

**Review by rooftopcellist — APPROVED** [review]

**Comment by ryanpetrello** [comment]
To clarify, this is fallout from the py3 merge.  @rooftopcellist noticed this code now fails on SCM checkouts w/ e.g., `.DS_Store` files.

**Review by AlanCoding — APPROVED** [review]
if it's not utf-8, we would need to know what to expect it is, and it's easiest to not get into that. This could be a documentation-worthy note.

File: awx/main/tests/unit/utils/ansible/playbooks/invalid/bad_encoding [hunk 2]
```diff
@@ -0,0 +1,2 @@
+�
```
Comment (ryanpetrello) [inline awx/main/tests/unit/utils/ansible/playbooks/invalid/bad_encoding:1]: @rooftopcellist this file contains `shift_jisx0213` encoded Japanese characters to make sure we don't regress on the issue you found.

**Comment by ryanpetrello** [comment]
@AlanCoding we didn't have any tests for these helper functions, so I decided to add some w/ some example files that pass/fail discovery.  Let me know what you think of this approach.

File: awx/main/utils/ansible.py [hunk 3]
```diff
@@ -44,13 +45,20 @@ def could_be_playbook(project_path, dir_path, filename):
...
+        )):
             if valid_playbook_re.match(line):
                 matched = True
+                break
```
Comment (ryanpetrello) [inline awx/main/utils/ansible.py:56]: If we find a match on one line, there's no need to read the _rest_ of the file; we can bail out at this point.

File: awx/main/utils/ansible.py [hunk 4]
```diff
@@ -44,13 +45,20 @@ def could_be_playbook(project_path, dir_path, filename):
...
             # Any YAML file can also be encrypted with vault;
             # allow these to be used as the main playbook.
             elif n == 0 and line.startswith('$ANSIBLE_VAULT;'):
                 matched = True
+                break
```
Comment (ryanpetrello) [inline awx/main/utils/ansible.py:61]: Same is true here.

**Comment by ryanpetrello** [comment]
recheck

**Comment by ryanpetrello** [comment]
recheck

**Comment by ryanpetrello** [comment]
recheck

**Comment by AlanCoding** [comment]
This is kind of a belated comment, but using the `awx/main/tests/unit/utils/ansible/` folder is inconsistent with how we've previously added data files in a common directory:

https://github.com/ansible/awx/tree/devel/awx/main/tests/data

This is on my mind because as I'm writing tests for inventory plugins, I'm adding new data files. Right now I'm planning on using a subdirectory in the common data folder.

**Comment by ryanpetrello** [comment]
Bah, I was looking for that folder and couldn't find it.

**Pull request merged**